terraso_backend/apps/soil_id/graphql/soil_id.py

```diff
@@ -251,65 +258,157 @@ def resolve_soil_match(soil_match):
+    # TODO: update cracks value when https://github.com/techmatters/soil-id-algorithm/pull/96 lands
+    # TODO: pass in values for elevation and bedrock
+    inputs = {
+        "soilHorizon": [],
+        "horizonDepth": [],
+        "rfvDepth": [],
+        "lab_Color": [],
+        "pSlope": data.slope,
+        "pElev": None,  # meters
+        "bedrock": None,
+        "cracks": False,
+    }
+
+    depths = data.depth_dependent_data
+    if len(depths) > 0 and depths[0].depth_interval.start != 0:
+        inputs["horizonDepth"].append(depths[0].depth_interval.end)
+        inputs["soilHorizon"].append(None)
+        inputs["rfvDepth"].append(None)
+        inputs["lab_Color"].append(None)
+        depths = depths[1:]
+
+    for depth in depths:
+        inputs["horizonDepth"].append(depth.depth_interval.end)
+        inputs["soilHorizon"].append(parse_texture(depth.texture))
+        inputs["rfvDepth"].append(parse_rock_fragment_volume(depth.rock_fragment_volume))
+        inputs["lab_Color"].append(parse_color_LAB(depth.color_LAB))
+
+    return inputs
+
+
 # to be replaced by actual algorithm output
 def resolve_data_based_soil_matches(
     _parent, _info, latitude: float, longitude: float, data: SoilIdInputData
 ):
-    return DataBasedSoilMatches(
-        matches=[
-            DataBasedSoilMatch(
-                data_source="SSURGO",
-                distance_to_nearest_map_unit_m=0.0,
-                location_match=SoilMatchInfo(score=1.0, rank=0),
-                data_match=SoilMatchInfo(score=0.2, rank=1),
-                combined_match=SoilMatchInfo(score=0.6, rank=1),
-                soil_info=sample_soil_infos[0],
-            ),
-            DataBasedSoilMatch(
-                data_source="STATSGO",
-                distance_to_nearest_map_unit_m=50.0,
-                location_match=SoilMatchInfo(score=0.5, rank=1),
-                data_match=SoilMatchInfo(score=0.75, rank=0),
-                combined_match=SoilMatchInfo(score=0.625, rank=0),
-                soil_info=sample_soil_infos[1],
-            ),
-        ]
+    list_result = list_soils(lat=latitude, lon=longitude)
+    if isinstance(list_result, str):
+        return None
+
+    result = rank_soils(
+        lat=latitude,
+        lon=longitude,
+        list_output_data=list_result,
+        **parse_rank_soils_input_data(data)
     )
 
+    try:
+        ranked_matches = []
+        for ranked_match in result["soilRank"]:
+            rankValues = [
+                ranked_match["rank_loc"],
+                ranked_match["rank_data"],
+                ranked_match["rank_data_loc"],
+            ]
+            if all([value != "Not Displayed" for value in rankValues]):
+                ranked_matches.append(
+                    resolve_data_based_soil_match(
+                        list_result.soil_list_json["soilList"], ranked_match
+                    )
+                )
+    except Exception:
+        logger.info(traceback.format_exc())
```

Comment:
question: can we use `logger.exception()` here instead? does that allow us to avoid importing `traceback`?

Reply:
ah whoops this was just for debugging, removed it